### PR TITLE
boilerplate introduction of the conversations api

### PR DESCRIPTION
**Pull request by ryanfkeepers**
adds basic get requests for the conversations api set to the api package.  Plust some other minor change for formatting, correctness, or necessity.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #4536 

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/pkg/backup/details/groups.go [hunk 1]
```diff
@@ -42,8 +42,11 @@ type GroupsInfo struct {
 	Modified time.Time `json:"modified,omitempty"`
 
 	// Channels Specific
-	Message   ChannelMessageInfo `json:"message"`
-	LastReply ChannelMessageInfo `json:"lastReply"`
+	Message   ChannelMessageInfo `json:"message,omitempty"`
+	LastReply ChannelMessageInfo `json:"lastReply,omitempty"`
+
+	// Conversations Specific
+	Post ConversationPostInfo `json:"post,omitempty"`
```
Comment (meain) [inline src/pkg/backup/details/groups.go:49]: 1) Can't we just reuse `ChannelMessageInfo` (maybe rename it)
1) Do we even need a separate field? Can we just fold into `Message` with `ItemType` deciding the type?

File: src/internal/operations/test/helper_test.go [hunk 2]
```diff
@@ -728,6 +728,7 @@ func verifyExtensionData(
 	case path.OneDriveService:
 		detailsSize = itemInfo.OneDrive.Size
 	case path.GroupsService:
+		// FIXME: needs update for message.
```
Comment (meain) [inline src/internal/operations/test/helper_test.go:731]: Should we create an issue to track this so that this does not get missed.

**Review by meain — APPROVED** [review]
Few thoughts, but approving to unblock.

File: src/pkg/backup/details/groups.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/pkg/backup/details/groups.go:49]: I don't see the value in it.  A little bit of duplication here won't hurt us, and it'll help minimize confusion for SDK consumers.  Per itemType, that's a largely internal reference.  End consumers don't have a good way of knowing which available value do and do not get populated based on type, nor are we providing strict guarantees of stability there.

File: src/internal/operations/test/helper_test.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/operations/test/helper_test.go:731]: It gets handled later.

**Pull request merged**